plugin/src/generators/new-lib/generator.ts

```diff
@@ -0,0 +1,146 @@
+import { formatFiles, Tree } from '@nx/devkit';
+import { tsquery } from '@phenomnomnominal/tsquery';
+import { prompt } from 'enquirer';
+import type { LibraryAngularVersionSupport, LibrarySupport } from 'libs/models';
+import { isPropertyAssignment, isStringLiteral } from 'typescript';
+import type { NewLibGeneratorSchema } from './schema';
+
+type VersionSupportPrompt = Omit<LibraryAngularVersionSupport, 'support'> & {
+  support: string;
+};
+
+export async function newLibGenerator(
+  tree: Tree,
+  options: NewLibGeneratorSchema
+) {
+  const normalizedOptions = normalizeOptions(options);
+  const libDataPath = 'src/app/libs.data.ts';
+  const libDataContent = tree.read(libDataPath, 'utf8');
+
+  if (!libDataContent) {
+    console.log(`${libDataPath} is empty. Cannot proceed`);
+    return;
+  }
+
+  const existingLibNameNodes = tsquery.query(
+    libDataContent,
+    `PropertyAssignment:has(Identifier[name=name])`
+  );
+  const exist = existingLibNameNodes.some((node) => {
```

Comment:
Could be `const exist = existingLibNameNodes.some((node) => 
  isPropertyAssignment(node) && isStringLiteral(node.initializer) && node.initializer.text === normalizedOptions.name
);`??

Reply:
Type Narrowing doesn't work after `isStringLiteral(node.initializer)` unfortunately but I don't mind doing `(node.initializer as ts.StringLiteral).text` either. No strong opinion on this one 🙂. If you have strong opinion about `as ts.StringLiteral`, I can change to that

Reply:
Switched to use query for the `StringLiteral` node so we can do `some(node => isStringLiteral(node) && node.text.includes(...))` instead.